client/ayon_flame/otio/flame_export.py

```diff
@@ -575,57 +635,44 @@ def create_otio_timeline(sequence):
             otio_track = create_otio_track(
                 "video", str(track.name)[1:-1])
 
-            all_segments = []
-            for segment in track.segments:
-                clip_data = _get_segment_attributes(segment)
-                if not clip_data:
-                    continue
-                all_segments.append(clip_data)
-
-            segments_ordered = dict(enumerate(all_segments))
-            log.debug("_ segments_ordered: {}".format(
-                pformat(segments_ordered)
-            ))
-            if not segments_ordered:
-                continue
-
-            for itemindex, segment_data in segments_ordered.items():
-                log.debug("_ itemindex: {}".format(itemindex))
-
-                # Add Gap if needed
-                prev_item = (
-                    segment_data
-                    if itemindex == 0
-                    else segments_ordered[itemindex - 1]
-                )
-                log.debug("_ segment_data: {}".format(segment_data))
-
-                # calculate clip frame range difference from each other
-                clip_diff = segment_data["record_in"] - prev_item["record_out"]
+            # Add segments onto track.
+            segments = _get_segments_from_track(track)
+            log.debug("_ segments: %s", pformat(segments))
+            _distribute_segments_on_track(segments, otio_track)
 
-                # add gap if first track item is not starting
-                # at first timeline frame
-                if itemindex == 0 and segment_data["record_in"] > 0:
-                    add_otio_gap(segment_data, otio_track, 0)
+            # add track to otio timeline
+            otio_timeline.tracks.append(otio_track)
 
-                # or add gap if following track items are having
-                # frame range differences from each other
-                elif itemindex and clip_diff != 1:
-                    add_otio_gap(
-                        segment_data, otio_track, prev_item["record_out"])
+    # create otio audio tracks with clips
+    for audio_track in sequence.audio_tracks:
 
-                # create otio clip and add it to track
-                otio_clip = create_otio_clip(segment_data)
-                otio_track.append(otio_clip)
+        if (
```

Comment:
I would revert this condition to first skip, log and continue to avoid the `else` at the end.

Reply:
Good call, this is done !